core/src/main/java/com/linecorp/armeria/common/Flags.java

```diff
@@ -88,279 +79,235 @@
-                CSV_SPLITTER.splitToList(getNormalized("preferredIpV4Addresses", "", unused -> true))
-                            .stream()
-                            .map(cidr -> {
-                                try {
-                                    return InetAddressPredicates.ofCidr(cidr);
-                                } catch (Exception e) {
-                                    logger.warn("Failed to parse a preferred IPv4: {}", cidr);
-                                }
-                                return null;
-                            })
-                            .filter(Objects::nonNull)
-                            .collect(toImmutableList());
-        switch (preferredIpV4Addresses.size()) {
-            case 0:
-                PREFERRED_IP_V4_ADDRESSES = null;
-                break;
-            case 1:
-                PREFERRED_IP_V4_ADDRESSES = preferredIpV4Addresses.get(0);
-                break;
-            default:
-                PREFERRED_IP_V4_ADDRESSES = inetAddress -> {
-                    for (Predicate<InetAddress> preferredIpV4Addr : preferredIpV4Addresses) {
-                        if (preferredIpV4Addr.test(inetAddress)) {
-                            return true;
-                        }
-                    }
-                    return false;
-                };
         }
     }
 
-    private static final boolean VERBOSE_SOCKET_EXCEPTIONS = getBoolean("verboseSocketExceptions", false);
+    private static final Sampler<Class<? extends Throwable>> VERBOSE_EXCEPTION_SAMPLER =
+            getValue(FlagsProvider::verboseExceptionSampler, "verboseExceptionSampler");
+
+    @Nullable
+    private static final Predicate<InetAddress> PREFERRED_IP_V4_ADDRESSES =
+            getValue(FlagsProvider::preferredIpV4Addresses, "preferredIpV4Addresses");
+
+    private static final boolean VERBOSE_SOCKET_EXCEPTIONS =
+            getValue(FlagsProvider::verboseSocketExceptions, "verboseSocketExceptions");
 
-    private static final boolean VERBOSE_RESPONSES = getBoolean("verboseResponses", false);
+    private static final boolean VERBOSE_RESPONSES =
+            getValue(FlagsProvider::verboseResponses, "verboseResponses");
 
     @Nullable
     private static final String REQUEST_CONTEXT_STORAGE_PROVIDER =
-            System.getProperty(PREFIX + "requestContextStorageProvider");
+            getValue(FlagsProvider::requestContextStorageProvider, "requestContextStorageProvider");
 
-    private static final boolean WARN_NETTY_VERSIONS = getBoolean("warnNettyVersions", true);
+    private static final boolean WARN_NETTY_VERSIONS =
+            getValue(FlagsProvider::warnNettyVersions, "warnNettyVersions");
 
-    private static final boolean USE_EPOLL = getBoolean("useEpoll", TransportType.EPOLL.isAvailable(),
+    private static final boolean DEFAULT_USE_EPOLL = TransportType.EPOLL.isAvailable();
+    private static final boolean USE_EPOLL = getBoolean("useEpoll",
```

Comment:
I didn't add `useEpoll()` to `FlagsProvider` because it was marked as deprecated (same as `annotatedServiceExceptionVerbosity()`). Should I add them to `FlagsProvider` or not?

Reply:
Ah, I think it's fine to leave unimplemented - I didn't realize they were deprecated 😅